Space side diff

TimvdLippe opened the pull request:
The space is now placed next to the diff view. When you click on the name of a block, you scroll to the corresponding diff on the right.

You can best test this PR with https://localhost:5200/projects/Polymer/Polymer/pulls/3680/diff

![image](https://cloud.githubusercontent.com/assets/5948271/15744177/bbe54656-28ca-11e6-855a-b3c887cee35d.png)

Fixes #168 
Fixes #178 
Fixes #184

TimvdLippe committed:
Initial rumblings of space next to diff

TimvdLippe committed:
Place space next to diff view

eanker commented:
![image](https://cloud.githubusercontent.com/assets/5946438/15744360/ce30b68c-28cb-11e6-8dcb-f923be1fba21.png)

As you already said, this doesn't work on small screens.

TimvdLippe committed:
Resize the container to make it work on smaller screens

TimvdLippe commented:
It should be a lot better now on your screen :)

txsmith commented on src/projects/pull-request/pull-request-page.html:
```diff
@@ -114,6 +113,11 @@
         margin: 0 auto;
         width: 75%;
       }
+      iron-pages[wide] {
```
I think we need a different solution for this `wide` mode. At this point, when I click the 'Show Space' button, the entire pull-request-view goes into wide-mode, making the comments and description section way to wide. I want the wide-mode to only be active if the space view is visible.

txsmith commented on src/projects/pull-request/pull-request-page.html:
```diff
@@ -159,22 +163,20 @@
-      <pull-request-space data-route="space" groups="[[orderedGroups]]"></pull-request-space>
+      <pull-request-changes data-route="diff" diff="[[diff]]" ordering="[[ordering]]"
+          base-url="[[selectedPull.head.repo.html_url]]" sha="[[selectedPull.head.sha]]"
+          show-space="{{spaceShown}}"></pull-request-changes>
```
Maybe property should be set to false as soon as the space not _visible_

txsmith commented:
When there is one chunk, and I click the `Link` button the linking-mode cannot be canceled, so I'll always see an arrow following the mouse pointer

txsmith commented on src/projects/pull-request/pull-request-page.html:
```diff
@@ -114,6 +113,11 @@
+      iron-pages[wide] {
+        max-width: 1800px;
+        width: 95%;
+        margin: 0 20px;
```
This margin makes the element render out of the center

TimvdLippe committed:
Fix width in changes instead of iron-pages

TimvdLippe committed:
Hit escape or click stop to stop creating a link

TimvdLippe commented:
You can now hit escape and the button text changes to indicate you can reclick to stop the creation of a connection :tada:

eanker commented:
It still does not work on my 13.3 inch screen.

![image](https://cloud.githubusercontent.com/assets/5946438/15788397/b3246062-29c7-11e6-9726-59678c64e426.png)

TimvdLippe committed:
Fix width regression

TimvdLippe commented:
Oops that was actually regressed while I was fixing the width calculations. It should be fixed now!

txsmith commented:
Can the space and diff views be centered?
![screenshot from 2016-06-06 10 35 34](https://cloud.githubusercontent.com/assets/5946422/15816305/73ea18da-2bd2-11e6-9ef1-4f900d9278c4.png)

txsmith commented:
When one navigation-list is shown, it slides over the space view. Is this intended?

txsmith commented:
Scrolling with space-view open in this PR is broken: https://localhost:5200/projects/Polymer/Polymer/pulls/3680/diff

TimvdLippe committed:
Center space and diff view

TimvdLippe committed:
Correct height of the space view

The width is calculated wrongly though, because it is not
positioned to the left

txsmith commented:
We'll do more fixes in another PR.
